js/main.js

```diff
@@ -0,0 +1,113 @@
+var $photoUrl = document.getElementById('photo-url');
+var $entryForm = document.querySelector('.entry-form-container');
+var $entryImage = document.getElementById('entry-image');
+var $newButton = document.getElementById('new-button');
+var $navBar = document.getElementById('nav-bar');
+
+document.addEventListener('DOMContentLoaded', function () {
+  if (data.nextEntryId > 1) {
+    var $navEntries = document.getElementById('nav-entries');
+    $navEntries.classList.remove('hidden');
+    var $entryList = document.getElementById('entry-list');
+    $entryList.innerHTML = '';
+    for (var i = data.entries.length - 1; i >= 0; i--) {
+      var entry = data.entries[i];
+      var $entry = renderEntry(entry);
+      $entryList.appendChild($entry);
+    }
+  } else {
+    document.getElementById('entry-list').innerHTML = '<p class="no-entries">No entries have been recorded.</p>';
+  }
+});
+
+$photoUrl.addEventListener('input', function (e) {
+  if ($entryImage.src === '') {
+    $entryImage.style.display = 'none';
+  } else {
+    $entryImage.src = e.target.value;
+  }
+});
+
+$entryForm.addEventListener('submit', function (e) {
+  e.preventDefault();
+  var newData = {
+    entryId: data.nextEntryId,
+    title: $entryForm.elements.entryTitle.value,
+    photoUrl: $entryForm.elements.photoUrl.value,
+    notesText: $entryForm.elements.entryNotes.value
+  };
+  data.entries.unshift(newData);
+  if (data.nextEntryId === 1) {
+    document.getElementById('entry-list').innerHTML = '';
+  }
+  document.getElementById('entry-list').appendChild(renderEntry(newData));
```

Comment:
you want to use the prepend method here to add the entry to the start of the ul

Reply:
Done.